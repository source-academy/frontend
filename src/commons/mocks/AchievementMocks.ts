import {
  AchievementGoal,
  AchievementItem,
  GoalType
} from '../../features/achievement/AchievementTypes';
import { AND, EventTypes } from '../../features/achievement/ExpressionTypes';

export const mockAchievements: AchievementItem[] = [
  {
    uuid: '0',
    title: 'Rune Master',
    xp: 100,
    isVariableXp: false,
    isTask: true,
    position: 1,
    prerequisiteUuids: ['2', '1'],
    goalUuids: ['0'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/rune-master.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Rune Master achievement'
    }
  },
  {
    uuid: '1',
    title: 'Beyond the Second Dimension',
    xp: 100,
    isVariableXp: false,
    deadline: new Date(2020, 7, 6, 12, 30, 0),
    isTask: false,
    position: 0,
    prerequisiteUuids: [],
    goalUuids: ['1'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/btsd.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Beyond the Second Dimension achievement'
    }
  },
  {
    uuid: '2',
    title: 'Colorful Carpet',
    xp: 100,
    isVariableXp: false,
    deadline: new Date(2020, 7, 8, 9, 0, 0),
    isTask: false,
    position: 0,
    prerequisiteUuids: [],
    goalUuids: ['2'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/colorful-carpet.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Colorful Carpet achievement'
    }
  },
  {
    uuid: '3',
    title: '',
    xp: 100,
    isVariableXp: false,
    isTask: false,
    position: 0,
    prerequisiteUuids: [],
    goalUuids: [],
    cardBackground: '',
    view: {
      coverImage: '',
      description: '',
      completionText: ''
    }
  },
  {
    uuid: '4',
    title: 'Curve Wizard',
    xp: 100,
    isVariableXp: false,
    deadline: new Date(2020, 8, 15, 0, 0, 0),
    isTask: true,
    position: 4,
    prerequisiteUuids: ['5', '6'],
    goalUuids: ['3'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/curve-wizard.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Curve Wizard achievement'
    }
  },
  {
    uuid: '5',
    title: 'Curve Introduction',
    xp: 100,
    isVariableXp: false,
    deadline: new Date(2020, 7, 28, 0, 0, 0),
    isTask: false,
    position: 0,
    prerequisiteUuids: [],
    goalUuids: ['4'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/curve-introduction.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Curve Introduction achievement'
    }
  },
  {
    uuid: '6',
    title: 'Curve Manipulation',
    xp: 100,
    isVariableXp: false,
    deadline: new Date(2020, 8, 5, 0, 0, 0),
    isTask: false,
    position: 0,
    prerequisiteUuids: [],
    goalUuids: ['5'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/curve-manipulation.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Curve Manipulation achievement'
    }
  },
  {
    uuid: '21',
    title: 'The Source-rer',
    xp: 100,
    isVariableXp: false,
    deadline: new Date(2020, 7, 21, 0, 0, 0),
    isTask: true,
    position: 3,
    prerequisiteUuids: [],
    goalUuids: ['16', '18'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/the-source-rer.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed The Source-rer achievement'
    }
  },
  {
    uuid: '8',
    title: 'Power of Friendship',
    xp: 100,
    isVariableXp: false,
    isTask: true,
    position: 2,
    prerequisiteUuids: ['9'],
    goalUuids: [],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/power-of-friendship.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Power of Friendship achievement'
    }
  },
  {
    uuid: '9',
    title: 'Piazza Guru',
    xp: 100,
    isVariableXp: false,
    isTask: false,
    position: 0,
    prerequisiteUuids: [],
    goalUuids: ['8'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/piazza-guru.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: 'Congratulations! You have completed Piazza Guru achievement'
    }
  },
  {
    uuid: '16',
    title: "That's the Spirit",
    xp: 100,
    isVariableXp: false,
    isTask: true,
    position: 5,
    prerequisiteUuids: [],
    goalUuids: ['14'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/thats-the-spirit.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud',
      completionText: "Congratulations! You have completed That's the Spirit achievement"
    }
  },
  {
    uuid: '13',
    title: 'Kool Kidz',
    xp: 100,
    isVariableXp: false,
    isTask: true,
    position: 6,
    prerequisiteUuids: [],
    goalUuids: ['11'],
    cardBackground:
      'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/card-background/default.png',
    view: {
      coverImage:
        'https://source-academy-assets.s3-ap-southeast-1.amazonaws.com/achievement/cover-image/default.png',
      description: 'Diz for teh K00L K1dz',
      completionText: 'U SO G00D'
    }
  }
];

export const mockGoals: AchievementGoal[] = [
  {
    uuid: '0',
    text: 'Bonus for completing Colorful Carpet & Beyond the Second Dimension Achievements',
    meta: {
      type: GoalType.BINARY,
      condition: AND(
        { event: EventTypes.ASSESSMENT_GRADING, restriction: 'M2A' },
        { event: EventTypes.ASSESSMENT_GRADING, restriction: 'M2B' }
      ),
      targetCount: 100
    },
    count: 0,
    targetCount: 100,
    completed: false,
    achievementUuids: ['0']
  },
  {
    uuid: '1',
    text: 'XP earned from Beyond the Second Dimension Mission',
    meta: {
      type: GoalType.ASSESSMENT,
      assessmentNumber: 1,
      requiredCompletionFrac: 0.5
    },
    count: 213,
    targetCount: 250,
    completed: true,
    achievementUuids: ['1']
  },
  {
    uuid: '2',
    text: 'XP earned from Colorful Carpet Mission',
    meta: {
      type: GoalType.ASSESSMENT,
      assessmentNumber: 2,
      requiredCompletionFrac: 0.8
    },
    count: 0,
    targetCount: 250,
    completed: false,
    achievementUuids: ['2']
  },
  {
    uuid: '3',
    text: 'Bonus for completing Curve Introduction & Curve Manipulation Achievements',
    meta: {
      type: GoalType.BINARY,
      condition: AND(
        {
          event: EventTypes.ASSESSMENT_GRADING,
          restriction: 'M3'
        },
        {
          event: EventTypes.ASSESSMENT_GRADING,
          restriction: 'M4A'
        }
      ),
      targetCount: 100
    },
    count: 0,
    targetCount: 100,
    completed: false,
    achievementUuids: ['4']
  },
  {
    uuid: '4',
    text: 'XP earned from Curve Introduction Mission',
    meta: {
      type: GoalType.ASSESSMENT,
      assessmentNumber: 3,
      requiredCompletionFrac: 0.5
    },
    count: 178,
    targetCount: 250,
    completed: true,
    achievementUuids: ['5']
  },
  {
    uuid: '5',
    text: 'XP earned from Curve Manipulation Mission',
    meta: {
      type: GoalType.ASSESSMENT,
      assessmentNumber: 4,
      requiredCompletionFrac: 0.8
    },
    count: 191,
    targetCount: 250,
    completed: false,
    achievementUuids: ['6']
  },
  {
    uuid: '16',
    text: 'Submit Source 3 Path',
    meta: {
      type: GoalType.BINARY,
      condition: {
        event: EventTypes.ASSESSMENT_SUBMISSION,
        restriction: 'P3'
      },
      targetCount: 100
    },
    count: 100,
    targetCount: 100,
    completed: true,
    achievementUuids: ['21']
  },
  {
    uuid: '18',
    text: 'XP earned from Source 3 Path',
    meta: {
      type: GoalType.ASSESSMENT,
      assessmentNumber: 5,
      requiredCompletionFrac: 1
    },
    count: 300,
    targetCount: 300,
    completed: true,
    achievementUuids: ['21']
  },
  {
    uuid: '8',
    text: 'Each Top Voted answer in Piazza gives 10 XP',
    meta: {
      type: GoalType.MANUAL,
      targetCount: 100
    },
    count: 40,
    targetCount: 100,
    completed: false,
    achievementUuids: ['9']
  },
  {
    uuid: '14',
    text: 'Submit 1 PR to Source Academy Github',
    meta: {
      type: GoalType.MANUAL,
      targetCount: 100
    },
    count: 100,
    targetCount: 100,
    completed: true,
    achievementUuids: ['16']
  },
  {
    uuid: '11',
    text: 'Be the Koolest Kidz in SOC by redeeming this 100 XP achievement yourself',
    meta: {
      type: GoalType.BINARY,
      condition: false,
      targetCount: 100
    },
    count: 0,
    targetCount: 100,
    completed: false,
    achievementUuids: ['13']
  }
];
