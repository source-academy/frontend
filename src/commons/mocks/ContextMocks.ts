import { parse } from 'acorn';
import { FunctionExpression, Node } from 'estree';
import { ACORN_PARSE_OPTIONS } from 'js-slang/dist/constants';
import createContext, { EnvTree } from 'js-slang/dist/createContext';
import Closure from 'js-slang/dist/interpreter/closure';
import { Context, Environment } from 'js-slang/dist/types';
import { TypeError } from 'js-slang/dist/utils/rttc';

export function mockContext(chapter = 1): Context {
  return createContext(chapter);
}

export function mockRuntimeContext(): Context {
  const context = createContext();
  // Note: noticed no harm in removing the following context.runtime.
  // If you get an error with head is undefined after trying to evaluate code.
  // Likely due to the environments: [].
  // In every real context, there is at least one env frame.
  context.runtime = {
    break: false,
    debuggerOn: true,
    isRunning: true,
    environmentTree: new EnvTree(),
    environments: [],
    nodes: [
      {
        type: 'Literal',
        loc: {
          start: { line: 1, column: 0 },
          end: { line: 1, column: 1 }
        },
        value: 0,
        raw: '0',
        range: [0, 1]
      }
    ],
    agenda: null,
    stash: null,
    envStepsTotal: 0,
    breakpointSteps: []
  };
  return context;
}

export function mockClosure(): Closure {
  return new Closure({} as FunctionExpression, {} as Environment, {} as Context);
}

export function mockTypeError(): TypeError {
  // Typecast to Node to fix estree-acorn compatability.
  return new TypeError(parse('', ACORN_PARSE_OPTIONS) as Node, '', '', '');
}
