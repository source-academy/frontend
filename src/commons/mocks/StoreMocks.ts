import _ from 'lodash';
import { DeepPartial, Store } from 'redux';
import mockStore from 'redux-mock-store';

import {
  defaultAcademy,
  defaultAchievement,
  defaultApplication,
  defaultDashboard,
  defaultFileSystem,
  defaultPlayground,
  defaultSession,
  defaultWorkspaceManager,
  OverallState
} from '../application/ApplicationTypes';

export function mockInitialStore(overrides?: DeepPartial<OverallState>): Store<OverallState> {
  const createStore = (mockStore as any)();
  const state: OverallState = {
    academy: defaultAcademy,
    achievement: defaultAchievement,
    application: defaultApplication,
    dashboard: defaultDashboard,
    playground: defaultPlayground,
    workspaces: defaultWorkspaceManager,
    session: defaultSession,
    fileSystem: defaultFileSystem
  };
  return createStore(_.merge(state, overrides));
}
