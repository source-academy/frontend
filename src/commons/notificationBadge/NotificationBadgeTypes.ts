export type Notification = {
  assessment_id?: number;
  assessment_type?: string;
  assessment_title?: string;
  id: number;
  submission_id?: number;
  type: NotificationType;
};

export enum NotificationTypes {
  new = 'new',
  deadline = 'deadline',
  autograded = 'autograded',
  graded = 'graded',
  submitted = 'submitted',
  unsubmitted = 'unsubmitted',
  new_message = 'new_message'
}

export type NotificationType = keyof typeof NotificationTypes;

export type NotificationFilterFunction = (notifications: Notification[]) => Notification[];
