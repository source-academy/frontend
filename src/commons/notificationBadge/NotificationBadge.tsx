import { Intent, PopoverInteractionKind, Position, Tag } from '@blueprintjs/core';
import { Popover2 } from '@blueprintjs/popover2';
import React from 'react';
import { useDispatch } from 'react-redux';

import { acknowledgeNotifications } from '../application/actions/SessionActions';
import { useTypedSelector } from '../utils/Hooks';
import { filterNotificationsById } from './NotificationBadgeHelper';
import { Notification, NotificationType, NotificationTypes } from './NotificationBadgeTypes';

type NotificationBadgeProps = OwnProps;

type OwnProps = {
  className?: string;
  disableHover?: boolean; // Set to true to disable popover content
  large?: boolean; // Set to true to use large style
  notificationFilter?: (notifications: Notification[]) => Notification[];
};

const NotificationBadge: React.SFC<NotificationBadgeProps> = props => {
  const dispatch = useDispatch();
  const initialNotifications = useTypedSelector(state => state.session.notifications);

  const notifications = props.notificationFilter
    ? props.notificationFilter(initialNotifications)
    : initialNotifications;

  if (!notifications.length) {
    return null;
  }

  const notificationIcon = (
    <Tag intent={Intent.DANGER} round={true} large={props.large}>
      {notifications.length}
    </Tag>
  );

  if (!props.disableHover) {
    const makeNotificationTag = (notification: Notification) => {
      const onRemove = () =>
        dispatch(acknowledgeNotifications(filterNotificationsById(notification.id)));

      return (
        <Tag
          className="row badge-tag"
          key={`${notification.id}`}
          minimal={true}
          multiline={true}
          onRemove={onRemove}
        >
          {makeNotificationMessage(notification.type)}
        </Tag>
      );
    };

    const notificationTags = <div className="col">{notifications.map(makeNotificationTag)}</div>;

    return (
      <Popover2
        className={props.className}
        content={notificationTags}
        interactionKind={PopoverInteractionKind.HOVER}
        placement={Position.RIGHT}
        hoverOpenDelay={50}
        hoverCloseDelay={50}
        lazy={true}
      >
        {notificationIcon}
      </Popover2>
    );
  }

  return notificationIcon;
};

const makeNotificationMessage = (type: NotificationType) => {
  switch (type) {
    case NotificationTypes.new:
      return 'This assessment is new.';
    case NotificationTypes.deadline:
      return 'This assessment is closing soon.';
    case NotificationTypes.autograded:
      return 'This assessment has been autograded.';
    case NotificationTypes.submitted:
      return 'This submission is new.';
    case NotificationTypes.unsubmitted:
      return 'This assessment has been unsubmitted.';
    case NotificationTypes.graded:
      return 'This assessment has been manually graded.';
    case NotificationTypes.new_message:
      return 'There are new messages.';
    default:
      return 'Unknown notification';
  }
};

export default NotificationBadge;
