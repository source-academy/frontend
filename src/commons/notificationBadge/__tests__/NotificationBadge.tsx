import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { mockInitialStore } from 'src/commons/mocks/StoreMocks';

import NotificationBadge from '../NotificationBadge';
import {
  filterNotificationsByAssessment,
  filterNotificationsBySubmission,
  filterNotificationsByType
} from '../NotificationBadgeHelper';
import { Notification } from '../NotificationBadgeTypes';

const notifications: Notification[] = [
  {
    id: 1,
    type: 'new',
    assessment_id: 1,
    assessment_type: 'Missions',
    assessment_title: 'The Secret to Streams'
  },
  {
    id: 2,
    type: 'graded',
    assessment_id: 1,
    assessment_type: 'Missions',
    assessment_title: 'The Secret to Streams'
  },
  {
    id: 3,
    type: 'autograded',
    assessment_id: 1,
    assessment_type: 'Missions',
    assessment_title: 'The Secret to Streams'
  },
  {
    id: 4,
    type: 'unsubmitted',
    assessment_id: 1,
    assessment_type: 'Missions',
    assessment_title: 'The Secret to Streams'
  }
];

describe('Badge', () => {
  test('renders properly with notifications', () => {
    const mockStore = mockInitialStore({ session: { notifications: notifications } });
    const tree = mount(
      <Provider store={mockStore}>
        <NotificationBadge />
      </Provider>
    );
    expect(tree.debug()).toMatchSnapshot();
  });

  test('does not render with no notifications', () => {
    const mockStore = mockInitialStore({ session: { notifications: [] } });
    const tree = mount(
      <Provider store={mockStore}>
        <NotificationBadge />
      </Provider>
    );

    expect(tree.debug()).toMatchSnapshot();
  });
});

describe('Badge with filter,', () => {
  test('filterNotificationsByAssessment renders properly', () => {
    const mockStore = mockInitialStore({ session: { notifications: notifications } });
    const tree = mount(
      <Provider store={mockStore}>
        <NotificationBadge notificationFilter={filterNotificationsByAssessment(1)} />
      </Provider>
    );
    expect(tree.debug()).toMatchSnapshot();
  });

  test('filterNotificationsBySubmission renders properly', () => {
    const mockStore = mockInitialStore({ session: { notifications: notifications } });
    const tree = mount(
      <Provider store={mockStore}>
        <NotificationBadge notificationFilter={filterNotificationsBySubmission(1)} />
      </Provider>
    );
    expect(tree.debug()).toMatchSnapshot();
  });

  test('filterNotificationsByAssessment renders properly', () => {
    const mockStore = mockInitialStore({ session: { notifications: notifications } });
    const tree = mount(
      <Provider store={mockStore}>
        <NotificationBadge notificationFilter={filterNotificationsByType('Missions')} />
      </Provider>
    );
    expect(tree.debug()).toMatchSnapshot();
  });
});
