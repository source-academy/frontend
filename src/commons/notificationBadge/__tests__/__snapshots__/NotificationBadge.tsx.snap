// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge does not render with no notifications 1`] = `
"<Provider store={{...}}>
  <NotificationBadge>
    <Blueprint4.Popover2 className={[undefined]} content={{...}} interactionKind=\\"hover\\" placement=\\"right\\" hoverOpenDelay={50} hoverCloseDelay={50} lazy={true} boundary=\\"clippingParents\\" captureDismiss={false} defaultIsOpen={false} disabled={false} fill={false} hasBackdrop={false} inheritDarkTheme={true} matchTargetWidth={false} minimal={false} openOnTargetFocus={true} positioningStrategy=\\"absolute\\" renderTarget={[undefined]} shouldReturnFocusOnClose={false} targetTagName=\\"span\\" transitionDuration={300} usePortal={true}>
      <Manager>
        <Reference>
          <Blueprint4.ResizeSensor2 targetRef={[Function (anonymous)]} onResize={[Function (anonymous)]}>
            <span aria-haspopup=\\"true\\" className=\\"bp4-popover2-target\\" onBlur={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
              <Blueprint4.Tag intent=\\"danger\\" round={true} large={[undefined]} className=\\"\\" disabled={[undefined]} tabIndex={0}>
                <span disabled={[undefined]} className=\\"bp4-tag bp4-intent-danger bp4-round\\" tabIndex={[undefined]}>
                  <Blueprint4.Icon icon={[undefined]} />
                  <Blueprint4.Text className=\\"bp4-fill\\" ellipsize={true} tagName=\\"span\\" title={[undefined]}>
                    <span className=\\"bp4-fill bp4-text-overflow-ellipsis\\" title={[undefined]}>
                      4
                    </span>
                  </Blueprint4.Text>
                  <Blueprint4.Icon icon={[undefined]} />
                </span>
              </Blueprint4.Tag>
            </span>
          </Blueprint4.ResizeSensor2>
        </Reference>
        <Popper innerRef={[Function (anonymous)]} placement=\\"right\\" strategy=\\"absolute\\" modifiers={{...}}>
          <Blueprint4.Overlay autoFocus={false} backdropClassName=\\"bp4-popover2-backdrop\\" backdropProps={{...}} canEscapeKeyClose={true} canOutsideClickClose={false} enforceFocus={true} hasBackdrop={false} isOpen={false} onClose={[Function (anonymous)]} onClosed={[undefined]} onClosing={[undefined]} onOpened={[undefined]} onOpening={[undefined]} transitionDuration={300} transitionName=\\"bp4-popover2\\" usePortal={true} portalClassName={[undefined]} portalContainer={[undefined]} shouldReturnFocusOnClose={false} lazy={true} />
        </Popper>
      </Manager>
    </Blueprint4.Popover2>
  </NotificationBadge>
</Provider>"
`;

exports[`Badge renders properly with notifications 1`] = `
"<Provider store={{...}}>
  <NotificationBadge>
    <Blueprint4.Popover2 className={[undefined]} content={{...}} interactionKind=\\"hover\\" placement=\\"right\\" hoverOpenDelay={50} hoverCloseDelay={50} lazy={true} boundary=\\"clippingParents\\" captureDismiss={false} defaultIsOpen={false} disabled={false} fill={false} hasBackdrop={false} inheritDarkTheme={true} matchTargetWidth={false} minimal={false} openOnTargetFocus={true} positioningStrategy=\\"absolute\\" renderTarget={[undefined]} shouldReturnFocusOnClose={false} targetTagName=\\"span\\" transitionDuration={300} usePortal={true}>
      <Manager>
        <Reference>
          <Blueprint4.ResizeSensor2 targetRef={[Function (anonymous)]} onResize={[Function (anonymous)]}>
            <span aria-haspopup=\\"true\\" className=\\"bp4-popover2-target\\" onBlur={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
              <Blueprint4.Tag intent=\\"danger\\" round={true} large={[undefined]} className=\\"\\" disabled={[undefined]} tabIndex={0}>
                <span disabled={[undefined]} className=\\"bp4-tag bp4-intent-danger bp4-round\\" tabIndex={[undefined]}>
                  <Blueprint4.Icon icon={[undefined]} />
                  <Blueprint4.Text className=\\"bp4-fill\\" ellipsize={true} tagName=\\"span\\" title={[undefined]}>
                    <span className=\\"bp4-fill bp4-text-overflow-ellipsis\\" title={[undefined]}>
                      4
                    </span>
                  </Blueprint4.Text>
                  <Blueprint4.Icon icon={[undefined]} />
                </span>
              </Blueprint4.Tag>
            </span>
          </Blueprint4.ResizeSensor2>
        </Reference>
        <Popper innerRef={[Function (anonymous)]} placement=\\"right\\" strategy=\\"absolute\\" modifiers={{...}}>
          <Blueprint4.Overlay autoFocus={false} backdropClassName=\\"bp4-popover2-backdrop\\" backdropProps={{...}} canEscapeKeyClose={true} canOutsideClickClose={false} enforceFocus={true} hasBackdrop={false} isOpen={false} onClose={[Function (anonymous)]} onClosed={[undefined]} onClosing={[undefined]} onOpened={[undefined]} onOpening={[undefined]} transitionDuration={300} transitionName=\\"bp4-popover2\\" usePortal={true} portalClassName={[undefined]} portalContainer={[undefined]} shouldReturnFocusOnClose={false} lazy={true} />
        </Popper>
      </Manager>
    </Blueprint4.Popover2>
  </NotificationBadge>
</Provider>"
`;

exports[`Badge with filter, filterNotificationsByAssessment renders properly 1`] = `
"<Provider store={{...}}>
  <NotificationBadge notificationFilter={[Function (anonymous)]}>
    <Blueprint4.Popover2 className={[undefined]} content={{...}} interactionKind=\\"hover\\" placement=\\"right\\" hoverOpenDelay={50} hoverCloseDelay={50} lazy={true} boundary=\\"clippingParents\\" captureDismiss={false} defaultIsOpen={false} disabled={false} fill={false} hasBackdrop={false} inheritDarkTheme={true} matchTargetWidth={false} minimal={false} openOnTargetFocus={true} positioningStrategy=\\"absolute\\" renderTarget={[undefined]} shouldReturnFocusOnClose={false} targetTagName=\\"span\\" transitionDuration={300} usePortal={true}>
      <Manager>
        <Reference>
          <Blueprint4.ResizeSensor2 targetRef={[Function (anonymous)]} onResize={[Function (anonymous)]}>
            <span aria-haspopup=\\"true\\" className=\\"bp4-popover2-target\\" onBlur={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
              <Blueprint4.Tag intent=\\"danger\\" round={true} large={[undefined]} className=\\"\\" disabled={[undefined]} tabIndex={0}>
                <span disabled={[undefined]} className=\\"bp4-tag bp4-intent-danger bp4-round\\" tabIndex={[undefined]}>
                  <Blueprint4.Icon icon={[undefined]} />
                  <Blueprint4.Text className=\\"bp4-fill\\" ellipsize={true} tagName=\\"span\\" title={[undefined]}>
                    <span className=\\"bp4-fill bp4-text-overflow-ellipsis\\" title={[undefined]}>
                      4
                    </span>
                  </Blueprint4.Text>
                  <Blueprint4.Icon icon={[undefined]} />
                </span>
              </Blueprint4.Tag>
            </span>
          </Blueprint4.ResizeSensor2>
        </Reference>
        <Popper innerRef={[Function (anonymous)]} placement=\\"right\\" strategy=\\"absolute\\" modifiers={{...}}>
          <Blueprint4.Overlay autoFocus={false} backdropClassName=\\"bp4-popover2-backdrop\\" backdropProps={{...}} canEscapeKeyClose={true} canOutsideClickClose={false} enforceFocus={true} hasBackdrop={false} isOpen={false} onClose={[Function (anonymous)]} onClosed={[undefined]} onClosing={[undefined]} onOpened={[undefined]} onOpening={[undefined]} transitionDuration={300} transitionName=\\"bp4-popover2\\" usePortal={true} portalClassName={[undefined]} portalContainer={[undefined]} shouldReturnFocusOnClose={false} lazy={true} />
        </Popper>
      </Manager>
    </Blueprint4.Popover2>
  </NotificationBadge>
</Provider>"
`;

exports[`Badge with filter, filterNotificationsByAssessment renders properly 2`] = `
"<Provider store={{...}}>
  <NotificationBadge notificationFilter={[Function (anonymous)]}>
    <Blueprint4.Popover2 className={[undefined]} content={{...}} interactionKind=\\"hover\\" placement=\\"right\\" hoverOpenDelay={50} hoverCloseDelay={50} lazy={true} boundary=\\"clippingParents\\" captureDismiss={false} defaultIsOpen={false} disabled={false} fill={false} hasBackdrop={false} inheritDarkTheme={true} matchTargetWidth={false} minimal={false} openOnTargetFocus={true} positioningStrategy=\\"absolute\\" renderTarget={[undefined]} shouldReturnFocusOnClose={false} targetTagName=\\"span\\" transitionDuration={300} usePortal={true}>
      <Manager>
        <Reference>
          <Blueprint4.ResizeSensor2 targetRef={[Function (anonymous)]} onResize={[Function (anonymous)]}>
            <span aria-haspopup=\\"true\\" className=\\"bp4-popover2-target\\" onBlur={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
              <Blueprint4.Tag intent=\\"danger\\" round={true} large={[undefined]} className=\\"\\" disabled={[undefined]} tabIndex={0}>
                <span disabled={[undefined]} className=\\"bp4-tag bp4-intent-danger bp4-round\\" tabIndex={[undefined]}>
                  <Blueprint4.Icon icon={[undefined]} />
                  <Blueprint4.Text className=\\"bp4-fill\\" ellipsize={true} tagName=\\"span\\" title={[undefined]}>
                    <span className=\\"bp4-fill bp4-text-overflow-ellipsis\\" title={[undefined]}>
                      4
                    </span>
                  </Blueprint4.Text>
                  <Blueprint4.Icon icon={[undefined]} />
                </span>
              </Blueprint4.Tag>
            </span>
          </Blueprint4.ResizeSensor2>
        </Reference>
        <Popper innerRef={[Function (anonymous)]} placement=\\"right\\" strategy=\\"absolute\\" modifiers={{...}}>
          <Blueprint4.Overlay autoFocus={false} backdropClassName=\\"bp4-popover2-backdrop\\" backdropProps={{...}} canEscapeKeyClose={true} canOutsideClickClose={false} enforceFocus={true} hasBackdrop={false} isOpen={false} onClose={[Function (anonymous)]} onClosed={[undefined]} onClosing={[undefined]} onOpened={[undefined]} onOpening={[undefined]} transitionDuration={300} transitionName=\\"bp4-popover2\\" usePortal={true} portalClassName={[undefined]} portalContainer={[undefined]} shouldReturnFocusOnClose={false} lazy={true} />
        </Popper>
      </Manager>
    </Blueprint4.Popover2>
  </NotificationBadge>
</Provider>"
`;

exports[`Badge with filter, filterNotificationsBySubmission renders properly 1`] = `
"<Provider store={{...}}>
  <NotificationBadge notificationFilter={[Function (anonymous)]} />
</Provider>"
`;
